### PR TITLE
G5 t7 recipe hydrator db

### PR DESCRIPTION
**Pull request by jdselby24**

### CODE REVIEW

File: src/Recipe/RecipeDBHydrator.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace GRUB\Recipe;
+use PDO;
+/**
+ * Class to create recipe entities from DB
+ */
+class RecipeDBHydrator {
```
Comment (sudokufan) [inline src/Recipe/RecipeDBHydrator.php]:
```suggestion
class RecipeDBHydrator
{
```

File: src/Recipe/RecipeDBHydrator.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        $this->$db  = $db;    
+    }
+
+    /**
+     * Function that retrives the recipe objects from the DB
+     *
+     * @return array Array of recipe objects
+     */
+    public function getRecipesFromDB() : array
```
Comment (sudokufan) [inline src/Recipe/RecipeDBHydrator.php]:
```suggestion
    public function getRecipesFromDB(): array
```

File: src/Recipe/RecipeDBHydrator.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    /**
+     * Function that retrives the recipe objects from the DB
+     *
+     * @return array Array of recipe objects
+     */
+    public function getRecipesFromDB(): array
+    {
+        $statement = "SELECT `title`,`link`,`imageURL`,`ingredients` FROM `recipes`;";
```
Comment (sudokufan) [inline src/Recipe/RecipeDBHydrator.php]:
```suggestion
        $statement = "SELECT `title`, `link`, `imageURL`, `ingredients` FROM `recipes`;";
```

File: src/Recipe/RecipeDBHydrator.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+     * @var PDO 
+     */
+    private $db;
+
+    /**
+     * Constructor that takes PDO object and stores in $db
+     *
+     * @param PDO $db
+     */
+    public function __construct(PDO $db)
```
Comment (sudokufan) [inline src/Recipe/RecipeDBHydrator.php:24]:
```suggestion

    public function __construct(PDO $db) 
```

File: src/Recipe/RecipeDBHydrator.php [hunk 5]
```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace GRUB\Recipe;
+use PDO;
+/**
```
Comment (sudokufan) [inline src/Recipe/RecipeDBHydrator.php:6]:
```suggestion

/**
```

Comment (sudokufan) [reply src/Recipe/RecipeDBHydrator.php:6]: improved spacing for readability

File: src/Recipe/RecipeDBHydrator.php [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+     *
+     * @param PDO $db
+     */
+    public function __construct(PDO $db) 
+    {
+        $this->$db  = $db;    
+    }
+
+    /**
+     * Function that retrives the recipe objects from the DB
```
Comment (bentleybiscuit) [inline src/Recipe/RecipeDBHydrator.php]:
```suggestion
     * Function that retrieves the recipe objects from the DB
```
typo

**Pull request merged**